baybe/searchspace/discrete.py

```diff
@@ -635,26 +635,22 @@ def get_candidates(
     def transform(
         self,
         data: pd.DataFrame,
+        allow_missing: bool = False,
     ) -> pd.DataFrame:
-        """Transform parameters from experimental to computational representation.
-
-        Continuous parameters and additional columns are ignored.
-
-        Args:
-            data: The data to be transformed. Must contain all specified parameters, can
-                contain more columns.
-
-        Returns:
-            A dataframe with the parameters in computational representation.
-        """
+        """See :func:`baybe.searchspace.core.SearchSpace.transform`."""
         # If the transformed values are not required, return an empty dataframe
         if self.empty_encoding or len(data) < 1:
             comp_rep = pd.DataFrame(index=data.index)
             return comp_rep
 
         # Transform the parameters
         dfs = []
-        for param in self.parameters:
+        parameters = (
```

Comment:
Note: I have NOT changed the old functionality (i.e., `allow_missing=False`) in order to have it fully backward-compatible. But I **do** agree that a better exception handling would have been nice even in the old code, so I'm happy to add it 👍🏼

Reply:
Will close this tread since it became duplicate. Let's continue the discussion [here](https://github.com/emdgroup/baybe/pull/278#discussion_r1656510663)